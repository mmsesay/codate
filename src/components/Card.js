import { Link } from 'react-router-dom';

export const Card = (prop) => {
  const {
    id, name, population, moreInfo,
  } = prop.data;

  return (
    <Link to={`/detail/${id}`}>
      <div className="text-lg p-3 h-52 flex flex-col items-end justify-between">
        <span className="p-1 text-sm">
          <i className="far fa-arrow-alt-circle-right fa-lg text-white"></i>
        </span>
        <div className="w-full h-28">
          <img 
            src={moreInfo.cityImageURL}
            alt="city preview image"
            className="w-full h-full rounded-md" />
        </div>
        <div className="text-right">
          <p>{name}</p>
          <p className="text-sm">{population.toLocaleString()}</p>
        </div>
      </div>
    </Link>
  );
};