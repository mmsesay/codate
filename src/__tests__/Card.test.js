import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import Card from '../components/Card';

const testData = {
  id: 1,
  name: 'Freetown',
  population: '7,000,000',
  moreInfo: {
    cityImage: 'test.png',
  },
};

describe('Testing Card component', () => {
  test('renders a message', () => {
    const { getByTestId } = render(<Card data={testData} />);
    expect(getByTestId('city-image')).toBeInTheDocument();
    expect(getByTestId('city-name')).toBeInTheDocument();
    expect(getByTestId('city-population')).toBeInTheDocument();
  });

  test('renders as expected', () => {
    const tree = renderer.create(<Card data={testData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
