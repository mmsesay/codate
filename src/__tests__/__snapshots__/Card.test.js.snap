// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Card component renders as expected 1`] = `
<div
  className="text-lg p-3 h-52 md:h-auto md:my-6 flex flex-col items-end justify-between"
>
  <span
    className="p-1 text-sm md:text-2xl md:mb-3"
  >
    <i
      className="far fa-arrow-alt-circle-right fa-lg text-white"
    />
  </span>
  <div
    className="w-full h-28 md:h-72"
  >
    <img
      alt="city preview image"
      className="w-full h-full rounded-md"
      data-testid="city-image"
    />
  </div>
  <div
    className="text-right text-white md:mt-4"
  >
    <p
      className="font-bold md:text-3xl"
      data-testid="city-name"
    >
      FREETOWN
    </p>
    <p
      className="text-sm md:text-xl"
      data-testid="city-population"
    >
      Population: 
      7,000,000
    </p>
  </div>
</div>
`;
